packages/cli/src/concurrency/concurrency-control.service.ts

```diff
@@ -35,52 +38,68 @@ export class ConcurrencyControlService {
 	) {
 		this.logger = this.logger.scoped('concurrency');
 
-		this.productionLimit = config.getEnv('executions.concurrency.productionLimit');
+		this.limits = new Map([
+			['production', config.getEnv('executions.concurrency.productionLimit')],
+			['evaluation', config.getEnv('executions.concurrency.evaluationLimit')],
+		]);
 
-		if (this.productionLimit === 0) {
-			throw new InvalidConcurrencyLimitError(this.productionLimit);
-		}
+		this.limits.forEach((limit, type) => {
+			if (limit === 0) {
+				throw new InvalidConcurrencyLimitError(limit);
+			}
 
-		if (this.productionLimit < -1) {
-			this.productionLimit = -1;
-		}
+			if (limit < -1) {
+				this.limits.set(type, -1);
+			}
+		});
 
-		if (this.productionLimit === -1 || config.getEnv('executions.mode') === 'queue') {
+		if (
+			Array.from(this.limits.values()).every((limit) => limit === -1) ||
+			config.getEnv('executions.mode') === 'queue'
+		) {
 			this.isEnabled = false;
 			return;
 		}
 
-		this.productionQueue = new ConcurrencyQueue(this.productionLimit);
+		this.queues = new Map();
+		this.limits.forEach((limit, type) => {
+			if (limit > 0) {
+				this.queues.set(type, new ConcurrencyQueue(limit));
+			}
+		});
 
 		this.logInit();
 
 		this.isEnabled = true;
 
-		this.productionQueue.on('concurrency-check', ({ capacity }) => {
-			if (this.shouldReport(capacity)) {
-				this.telemetry.track('User hit concurrency limit', {
-					threshold: CLOUD_TEMP_PRODUCTION_LIMIT - capacity,
-				});
-			}
-		});
-
-		this.productionQueue.on('execution-throttled', ({ executionId }) => {
-			this.logger.debug('Execution throttled', { executionId });
-			this.eventService.emit('execution-throttled', { executionId });
-		});
-
-		this.productionQueue.on('execution-released', async (executionId) => {
-			this.logger.debug('Execution released', { executionId });
+		this.queues.forEach((queue, type) => {
+			queue.on('concurrency-check', ({ capacity }) => {
+				if (this.shouldReport(capacity)) {
+					this.telemetry.track('User hit concurrency limit', {
+						threshold: CLOUD_TEMP_PRODUCTION_LIMIT - capacity,
+						concurrencyType: type,
+					});
+				}
+			});
```

Comment:
Important - To double check, have you cleared with JC that it is fine to change the existing telemetry call like this? 

This call was meant specifically for the production concurrency limit, in order to find out reasonable thresholds.

Reply:
Thanks for pointing that! I checked with JC, it's ok, I just need to add the new prop to the docs